go/pkg/client/tree.go

```diff
@@ -227,45 +230,65 @@ func evalParentSymlinks(execRoot, relPath string, materializeOutsideExecRoot boo
 	return evaledPathBuilder.String(), symlinks, nil
 }
 
+// loadIntermediateSymlinks inserts symlink nodes into fs.
+// If the symlink path already exists in fs, it is not overwritten, which means the first entry wins.
+// This helps avoid redundant allocations from shared ancestors.
```

Comment:
Do you mind adding an example here? it helps with reasoning :)

Reply:
Done